azkaban-exec-server/src/main/java/azkaban/execapp/JobRunner.java

```diff
@@ -845,6 +856,21 @@ private boolean prepareJob() throws RuntimeException {
     return true;
   }
 
+  // If linux group and user needs to be added for effectiveUser before job process starts
+  // then set it up before effective user is used to set permission for any file.
+  private void addGroupAndUser(final Props pluginPrivateProps) throws Exception {
```

Comment:
Another concern:
With this mechanism in place it's possible that different instances of jobs will see different user-id for the same user, which isn't the case when using LDAP + nscd. 
If for any reason a job has made assumptions about the UID for a user being same across all invocations (or matching LDAP)  then it could run into trouble. 
I don't think there should be many (if any) jobs making this assumption but still wanted to highlight the possibility.

Reply:
Oh yes. I tested couple of flows and did not see any assumption related to same uid. Actually that was the motivation behind adding this as a part of azkaban code rather than finding generic solution.